components/match2/wikis/battlerite/match_group_input_custom.lua

```diff
@@ -51,378 +28,162 @@ local CustomMatchGroupInput = {}
-		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
-		match.finished = true
-	elseif Logic.isNumeric(match.winner) and Table.includes(ALLOWED_STATUSES, match.walkover) then
-		local winnerIndex = tonumber(match.winner)
-		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
-		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
-		match.finished = true
-	end
-
-	-- see if match should actually be finished if score is set
-	if isScoreSet and not Logic.readBool(match.finished) and match.timestamp ~= DateExt.defaultTimestamp then
-		local threshold = match.dateexact and 30800 or 86400
-		if match.timestamp + threshold < NOW then
-			match.finished = true
-		end
-	end
-
-	-- apply placements and winner if finshed
-	if
-		not Logic.isEmpty(match.winner) or
-		Logic.readBool(match.finished) or
-		CustomMatchGroupInput.placementCheckSpecialStatus(opponents)
-	then
-		match.finished = true
-		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
-	end
-
-	-- Update all opponents with new values
-	for opponentIndex, opponent in pairs(opponents) do
-		match['opponent' .. opponentIndex] = opponent
-	end
-	return match
-end
-
----@param opponents table[]
----@param walkoverType string?
----@return table[]
-function matchFunctions._makeAllOpponentsLoseByWalkover(opponents, walkoverType)
-	for index, _ in pairs(opponents) do
-		opponents[index].score = NOT_PLAYED_SCORE
-		opponents[index].status = walkoverType
-	end
-	return opponents
 end
 
 --
 -- map related functions
 --
 
--- Parse extradata information
+-- Check if a map should be discarded due to being redundant
+-- DUMMY_MAP_NAME needs the match the default value in Template:Map
 ---@param map table
----@return table
-function mapFunctions.getExtraData(map)
-	map.extradata.comment = map.comment
-
-	return map
+---@return boolean
+function MapFunctions.keepMap(map)
+	return map.map ~= DUMMY_MAP
 end
 
--- Calculate Score and Winner of the map
 ---@param map table
+---@param opponentCount integer
 ---@return table
-function mapFunctions.getScoresAndWinner(map)
-	map.scores = {}
-	local indexedScores = {}
-	for scoreIndex = 1, MAX_NUM_OPPONENTS do
-		-- read scores
-		local score = map['score' .. scoreIndex] or map['t' .. scoreIndex .. 'score']
-		local obj = {}
-		if not Logic.isEmpty(score) then
-			if Logic.isNumeric(score) then
-				obj.status = STATUS_SCORE
-				score = tonumber(score)
-				map['score' .. scoreIndex] = score
-				obj.score = score
-			elseif Table.includes(ALLOWED_STATUSES, score) then
-				obj.status = score
-				obj.score = NOT_PLAYED_SCORE
-			end
-			table.insert(map.scores, score)
-			indexedScores[scoreIndex] = obj
-		else
-			break
+function MapFunctions.getExtraData(map, opponentCount)
+	return {
+		comment = map.comment,
+	}
+end
+
+---@param winnerInput string|integer|nil
+---@param finished boolean
+---@return fun(opponentIndex: integer): integer?
+function MapFunctions.calculateMapScore(winnerInput, finished)
```

Comment:
Same, map func doesn't match the old one

Reply:
same as on arenafps